integrations/access/common/status.go

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+
+	"github.com/gravitational/teleport/api/types"
+)
+
+// StatusSink defines a destination for PluginStatus
+type StatusSink interface {
```

Comment:
This has an implementation in `e/`, so I think it should stay exporter? And I'd rather keep it in `common` since plugins other than Slack will use it in the long run.

Reply:
This is a good use of interfaces here, but I think exporting it will make it go against the usual interface advice: https://github.com/golang/go/wiki/CodeReviewComments#interfaces.